ScrapeRepos.ts

```diff
@@ -0,0 +1,356 @@
+ * @param {string} repo Repository name
+ * @param {string} authToken AuthToken for GitHub API
+ * @returns {[date: 'DATE', count: 'COUNT']}: { date: '2023-05-10', count: 1549 } 
+ * => on error empty array and error message
+ */
+async function fetchRepositoryStars(owner: string, repo: string, authToken: string) {
+  const query = `
+    query ($owner: String!, $repo: String!, $cursor: String) {
+      repository(owner: $owner, name: $repo) {
+        stargazers(first: 100, orderBy: { field: STARRED_AT, direction: DESC }, after: $cursor) {
+          totalCount
+          edges {
+            starredAt
+            node {
+              login
+            }
+          }
+          pageInfo {
+            endCursor
+            hasNextPage
+          }
+        }
+      }
+    }
+  `;
+
+  const headers = {
+    'Authorization': authToken
+  };
+
+  let hasNextPage = true;
+  let endCursor: string | null = null;
+
+  let starHistoryList: any[] = []
+
+  try {
+    while (hasNextPage) {
+      const variables: any = {
+        owner,
+        repo,
+        cursor: endCursor
+      };
+
+      const response = await axios.post(
+        'https://api.github.com/graphql',
+        { query, variables },
+        { headers }
+      );
+      const starsData = response.data.data.repository.stargazers.edges;
+      starHistoryList.push(starsData)
+
+      const pageInfo = response.data.data.repository.stargazers.pageInfo;
+      hasNextPage = pageInfo.hasNextPage;
+      endCursor = pageInfo.endCursor || null;
+
+      variables.cursor = endCursor;
+    }
+  } catch (error) {
+    console.error('Error:', error);
+    return []
+  }
+
+  // get the real star history
+  const mergedArray = [].concat(...starHistoryList);
+  const result = await starHistory(mergedArray);
+  return result
+}
+  
+
+/** Transforms an array of all the logged stars timestamps into a merged array with the star History
+ * @param {[starredAt: 'TIME', node: {Object} ]} starsData Array with the timestamps of the star history
+ * @returns {[date: 'DATE', count: 'COUNT']}:  { date: '2023-05-13', count: 515 }
+ */
+async function starHistory(starsData: any[]) {
+  const dateCounts: { [date: string]: number } = {};
+  starsData.forEach((item) => {
+    const date = item.starredAt.split('T')[0]; // Extract date portion
+    dateCounts[date] = (dateCounts[date] || 0) + 1; // Increment count for the date
+  });
+
+  // Convert dateCounts object into an array of objects
+  const result = Object.keys(dateCounts).map((date) => ({
+    date,
+    count: dateCounts[date]
+  }));
+
+  return result;
+}  
+
+
+/** Get trending developers (and their trending repos) from the github page
+ * @param {number} userInput describes the timeframe; 0: daily, 1: weekly; 2: monthly
+ * @returns   list of {name: 'NAME', username: 'USERNAME', repo: 'REPO'}
+ */
+async function fetchDevelopers(userInput: number){
+  const timeModes = ['daily', 'weekly', 'monthly'];
+
+  interface Developer {
```

Comment:
Please don't define an interface inside a function. If possible define the interfaces into a new file.